drivers/base/nmidebug/nmidebug.c

```diff
@@ -13,49 +12,31 @@
 
 /* FUNCTIONS ******************************************************************/
 
-PCHAR NmiBegin = "NMI4NMI@";
-
-FORCEINLINE
-VOID
-NmiClearFlag(VOID)
-{
-    ((PCHAR)&KiBugCheckData[4])[0] -= (NmiBegin[3] | NmiBegin[7]);
-    ((PCHAR)&KiBugCheckData[4])[3] |= 1;
-#ifdef _M_IX86
-#if defined(_MSC_VER) && !defined(__clang__)
-    __asm
-    {
-        rcr KiBugCheckData[4], 8
-    }
-#else
-    __asm__("rcrl %b[shift], %k[retval]" : [retval] "=rm" (KiBugCheckData[4]) : "[retval]" (KiBugCheckData[4]), [shift] "Nc" (8));
-#endif
-#endif
-}
-
 BOOLEAN
 NTAPI
-NmiDbgCallback(IN PVOID Context,
-               IN BOOLEAN Handled)
+NmiDbgCallback(
+    _In_opt_ PVOID Context,
+    _In_ BOOLEAN Handled)
 {
-    /* Clear the NMI flag */
-    NmiClearFlag();
+    /* Handle the NMI safely if a debugger is connected */
+    if (!KD_DEBUGGER_NOT_PRESENT && KD_DEBUGGER_ENABLED)
+    {
+        DbgPrint("NMI Callback entered!\n");
 
-    /* Get NMI status signature */
-    __indwordstring(0x80, (PULONG)NmiBegin, 1);
```

Comment:
This reads *one* (1) dword from port 0x80, which is (on PT-AT compatibles) reading the port for POST error codes (see https://stackoverflow.com/questions/6793899/what-does-the-0x80-port-address-connect-to ).
It stores it in the first 4 bytes in the NmiBegin buffer (erases the first "NMI4" part).

Or... is it used as "dummy" port for delay? (see https://groups.google.com/g/linux.kernel/c/UTKlqyBFiyU )

Reply:
>Or... is it used as "dummy" port for delay?

or to crash the system at certain condition...